### DISCUSSION TITLE
installation with desktop firewall is impossible

### DISCUSSION THREAD

**Discussion by jkischel** [LMS-Community/slimserver#1243] [General]
I just downloaded the 9.0.0 version for Windows. And yes, I have a desktop firewall which normally works perfectly.

Now, the following happens:
- start the setup
- confirm the language
- click on install
--> my firewall asks me if I want to allow this connection which I confirm.
- the setup ends and tells me that the connection cannot be established

so I start the setup again, and get the same result. and again and again and again.

The bug is that the installer which asks for a connection is always started from a different location and even gives up at once the connection was tried! How will it ever be possible to sensefully configure a working firewall rule for that?

**Comment by michaelherger** [comment]
Could you please share more details about the messages you get? What is asking for which application? Is this about accessing LMS on port 9000? About the installer reaching to the internet to download some component?

**Comment by jkischel** [comment]
Yes, sure.

This is a pure setup issue which will try to download the Strawberry Perl environment which fails because the responsible component of the installer which tries to establish the connection always will be created somewhere in temp folder, but each time a different location. With a desktop firewall that asks the user if he wants to allow that connection it's impossible to complete the setup as on the next trial, the path is different so the just set firewall rule does not apply.

I see two points here that would solve that issue:

1. The setup doesn't wait until the connection has been allowed or rejected, it immediately just fails. As there are other setups that would wait until the user makes a decision, this waiting would solve the problem. (just wanted to point out that in general, a connection establishment can obviously be put somehow 'on hold')

2. The setup just offers a one-shot and doesn't offer a 'retry' option. The user has no chance to "tell" the setup: that problem has been fixed, just try again. I think, also such a retry option would solve the issue as long as it's the same file what's trying to access the internet.

**Comment by michaelherger** [comment]
There's no bug in the installer. If a firewall wants to block a connection, then that's a problem. But that's nothing the installer can change, or the firewall would be useless. So you have to tell the firewall to just sit quiet for a few minutes and resume work once you're done?
